extensions-contrib/opentelemetry-extensions/src/main/java/org/apache/druid/data/input/opentelemetry/protobuf/traces/OpenTelemetryTracesProtobufInputFormat.java

```diff
@@ -0,0 +1,234 @@
+  private final String statusMessageDimension;
+  private final String kindDimension;
+
+  @JsonProperty
+  public String getSpanNameDimension()
+  {
+    return spanNameDimension;
+  }
+
+  @JsonProperty
+  public String getResourceAttributePrefix()
+  {
+    return resourceAttributePrefix;
+  }
+
+  @JsonProperty
+  public String getSpanIdDimension()
+  {
+    return spanIdDimension;
+  }
+
+  @JsonProperty
+  public String getParentSpanIdDimension()
+  {
+    return parentSpanIdDimension;
+  }
+
+  @JsonProperty
+  public String getTraceIdDimension()
+  {
+    return traceIdDimension;
+  }
+
+  @JsonProperty
+  public String getEndTimeDimension()
+  {
+    return endTimeDimension;
+  }
+
+  @JsonProperty
+  public String getStatusCodeDimension()
+  {
+    return statusCodeDimension;
+  }
+
+  @JsonProperty
+  public String getStatusMessageDimension()
+  {
+    return statusMessageDimension;
+  }
+
+  @JsonProperty
+  public String getKindDimension()
+  {
+    return kindDimension;
+  }
+
+  @JsonProperty
+  public String getSpanAttributePrefix()
+  {
+    return spanAttributePrefix;
+  }
+
+  private String validateDimensionName(String input, String dimensionName)
+  {
+    Preconditions.checkArgument(!input.isEmpty(),
+                                dimensionName + " dimension cannot be empty");
+
+    if (!resourceAttributePrefix.isEmpty()) {
+      Preconditions.checkArgument(!input.startsWith(resourceAttributePrefix),
+                                  " dimension name cannot start with resourceAttributePrefix");
+    }
+
+    if (!spanAttributePrefix.isEmpty()) {
+      Preconditions.checkArgument(!input.startsWith(spanAttributePrefix),
+                                  " dimension name cannot start with spanAttributePrefix");
+    }
+    return input;
+  }
+
+  public OpenTelemetryTracesProtobufInputFormat(
+      @JsonProperty("spanAttributePrefix") String spanAttributePrefix,
+      @JsonProperty("resourceAttributePrefix") String resourceAttributePrefix,
+      @JsonProperty("spanNameDimension") String spanNameDimension,
+      @JsonProperty("spanIdDimension") String spanIdDimension,
+      @JsonProperty("parentSpanIdDimension") String parentSpanIdDimension,
+      @JsonProperty("traceIdDimension") String traceIdDimension,
+      @JsonProperty("endTimeDimension") String endTimeDimension,
+      @JsonProperty("statusCodeDimension") String statusCodeDimension,
+      @JsonProperty("statusMessageDimension") String statusMessageDimension,
+      @JsonProperty("kindDimension") String kindDimension
+  )
+  {
+
+    this.spanAttributePrefix = spanAttributePrefix == null ? DEFAULT_SPAN_ATTR_PREFIX : spanAttributePrefix;
+    this.resourceAttributePrefix = resourceAttributePrefix == null ? DEFAULT_RESOURCE_ATTR_PREFIX : resourceAttributePrefix;
+
+    this.spanNameDimension = spanNameDimension == null ? DEFAULT_SPAN_NAME_DIMENSION :
+                             validateDimensionName(spanNameDimension, "Span Name");
```

Comment:
rather than using a different name in the error message, we should use the actual field name to make it clear which field is the problem.